pkg/apis/servicecatalog/types.go

```diff
@@ -266,10 +266,16 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// Parameters is set of properties to be
 	// passed to the underlying broker.
 	Parameters *runtime.RawExtension
 
+	// Optional: AlphaSecretParameters is reference to the secret object containing
+	// sensitive information to pass to the underlying broker. This may be
+	// empty if no secret object is specified.
+	// +optional
+	AlphaSecretParameters *v1.LocalObjectReference
```

Comment:
@pmorie let's have a look at the example:
```json
{
  "key1": "string",
  "key2": {
    "json": true,
    "childKey1": "string"
  },
  "key3": "{ json: false, childKey1: \"it's embedded json passed as a string\" }"
}
```
With separate `SecretParameter` object as you suggest, it's simply impossible to support such parameters structure. You have to decide upfront whether the parameter value has to **always** be a raw value or a JSON.